src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,116 @@
+                    continue;
+                }
+                int tank = 0;
+                if (chemicalHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                    //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                    tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % chemicalHandler.getTanks();
+                }
+                renderBar(bufferBuilder, xOffset, yOffset, chemicalHandler.getChemicalInTank(tank), chemicalHandler.getTankCapacity(tank));
+                yOffset--;
+            }
+        }
+
+        if (showFluid) {
+            Optional<IFluidHandlerItem> capabilityInstance = FluidUtil.getFluidHandler(stack).resolve();
+            if (capabilityInstance.isPresent()) {
+                IFluidHandlerItem fluidHandler = capabilityInstance.get();
+                if (fluidHandler.getTanks() != 0) {
+                    int tank = 0;
+                    if (fluidHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                        //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                        tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % fluidHandler.getTanks();
+                    }
+                    renderBar(bufferBuilder, xOffset, yOffset, fluidHandler.getFluidInTank(tank), fluidHandler.getTankCapacity(tank));
+                }
+            }
+        }
+        return true;
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, ChemicalStack<?> stack, long capacity) {
+        int color = stack.getChemicalColorRepresentation();
+        renderBar(bufferBuilder, stackXPos, yPos, (float)((double) stack.getAmount() / capacity), color);
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, FluidStack stack, long capacity) {
+        int color = IClientFluidTypeExtensions.of(stack.getFluid()).getTintColor(stack);
+        renderBar(bufferBuilder, stackXPos, yPos, (float)((double) stack.getAmount() / capacity), color);
+    }
+
+    private void renderBar(BufferBuilder bufferBuilder, int stackXPos, int yPos, float width, int color) {
+        ItemRenderer itemRenderer = Minecraft.getInstance().getItemRenderer();
+        itemRenderer.fillRect(bufferBuilder, stackXPos + 2, yPos, 13, 1, 0, 0, 0, 255);
+        itemRenderer.fillRect(bufferBuilder, stackXPos + 2, yPos, convertWidth(width), 1, getRed(color), getGreen(color), getBlue(color), 255);
+    }
+
+    private int convertWidth(double width) {
+        return MathUtils.clampToInt(Math.round(13.0F * width));
+    }
+
+    private int getRed(int color) {
+        return color >> 16 & 0xFF;
+    }
+
+    private int getGreen(int color) {
+        return color >> 8 & 0xFF;
+    }
+
+    private int getBlue(int color) {
+        return color & 0xFF;
+    }
```

Comment:
Use helpers from `MekanismRenderer` instead

Reply:
I've seen them but deemed them "unusuable", because they return a float divided by 255. I would have to multiply that value back and it's not worth it imo